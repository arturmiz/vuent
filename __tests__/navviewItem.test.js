import { mount } from '@vue/test-utils';
import VntNavviewItem from '@/components/navview/MenuItem.vue';

describe('NavviewItem', () => {
  let wrapper;

  const linkSelector = '.vnt-navview__item-link';

  describe('by default', () => {

    beforeAll(() => {
      wrapper = mount(VntNavviewItem);
    });

    test('renders correctly', () => {
      expect(wrapper).toMatchSnapshot();
    });

    test('is not active', () => {
      expect(wrapper.vm.active).toBe(false);
    });

    test('icon is null', () => {
      expect(wrapper.vm.icon).toBeNull();
    });

    test('click handler is undefined', () => {
      expect(wrapper.vm.click).toBeUndefined();
    });

    test('when clicked, no error is thrown', () => {
      expect(() => {
        wrapper.find(linkSelector).trigger('click');
      }).not.toThrow();
    });

  });

  describe('when icon is set', () => {

    beforeAll(() => {
      wrapper = mount(VntNavviewItem, {
        propsData: {
          icon: 'settings'
        },
        slots: {
          default: 'Settings'
        }
      });
    });

    test('renders correctly', () => {
      expect(wrapper).toMatchSnapshot();
    });

  });

  describe('when click handler is given', () => {
    let mockClick;

    beforeAll(() => {
      mockClick = jest.fn();

      wrapper = mount(VntNavviewItem, {
        propsData: {
          click: mockClick
        }
      });
    });

    test('when clicked, click handler is invoked', () => {
      wrapper.find(linkSelector).trigger('click');
      expect(mockClick).toHaveBeenCalled();
    });

  });

});
