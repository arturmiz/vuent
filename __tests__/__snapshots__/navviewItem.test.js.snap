// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavviewItem by default renders correctly 1`] = `
<li class="vnt-navview__item">
  <a href="#" class="vnt-navview__item-link"><span class="vnt-icon vnt-navview__item-icon"></span> <span class="vnt-navview__item-label"></span></a>
</li>
`;

exports[`NavviewItem when icon is set renders correctly 1`] = `
<li class="vnt-navview__item">
  <a href="#" class="vnt-navview__item-link"><span class="vnt-icon vnt-navview__item-icon vnt-icon--settings"></span> <span class="vnt-navview__item-label">Settings</span></a>
</li>
`;
